### PR TITLE
ARGO-220 Tenants api v2

### PR DESCRIPTION
**Pull request by kaggis**
Implemented CRUD requests for tenants resource 

**C**reate: `POST /api/v2/admin/tenants`
**R**ead_all: `GET /api/v2/admin/tenants`
**R**ead_one: `GET /api/v2/admin/tenants/{UUID}`
**U**pdate: `PUT /api/v2/admin/tenants/{UUID}`
**D**elete: `DELETE /api/v2/admin/tenants`

Tenant resource is described in the following json
```json

{
  "id": "6ac7d684-1f8e-4a02-a502-720e8f11e50b",
  "info": {
    "name": "Tenant1",
    "email": "email1@tenant1.com",
    "website": "www.tenant1.com",
    "created": "2015-10-20 02:08:04",
    "updated": "2015-10-20 02:08:04"
  },
  "db_conf": [
    {
      "store": "ar",
      "server": "a.mongodb.org",
      "port": 27017,
      "database": "ar_db",
      "username": "admin",
      "password": "3NCRYPT3D"
    },
    {
      "store": "status",
      "server": "b.mongodb.org",
      "port": 27017,
      "database": "status_db",
      "username": "admin",
      "password": "3NCRYPT3D"
    }
  ],
  "users": [
    {
      "name": "cap",
      "email": "cap@email.com",
      "api_key": "C4PK3Y"
    },
    {
      "name": "thor",
      "email": "thor@email.com",
      "api_key": "TH0RK3Y"
    }
  ]
}

```

- Unit test Added
- Updated tenant seed data through all packages (in unit tests)
- Added mkdoc

### CODE REVIEW

**Comment by kaggis** [comment]
please review 
/cc @makistsantekidis @pkoro

**Comment by makistsantekidis** [comment]
+1

File: app/tenants/controller.go [hunk 1]
```diff
@@ -52,38 +57,36 @@ func Create(r *http.Request, cfg config.Config) (int, http.Header, []byte, error
...
 
-			code = http.StatusBadRequest
-			h.Set("Content-Type", fmt.Sprintf("%s; charset=%s", contentType, charset))
-			return code, h, output, err
-		}
+	// Parse body json
+	if err := json.Unmarshal(body, &incoming); err != nil {
+
+		output, _ = respond.MarshalContent(respond.BadRequestBadJSON, contentType, "", " ")
+		code = 400
```
Comment (pkoro) [inline app/tenants/controller.go]: Maybe it's better to use the `http.StatusBadRequest` here.

File: app/tenants/controller.go [hunk 2]
```diff
@@ -95,56 +98,31 @@ func Create(r *http.Request, cfg config.Config) (int, http.Header, []byte, error
...
 	if err != nil {
-		code = http.StatusInternalServerError
-		return code, h, output, err
+		panic(err)
 	}
 
-	h.Set("Content-Type", fmt.Sprintf("%s; charset=%s", contentType, charset))
+	// Create view of the results
+	output, err = createRefView(incoming, "Tenant was succesfully created", 201, r) //Render the results into JSON
+	code = 201
```
Comment (pkoro) [inline app/tenants/controller.go]: Same comment on these two lines.

File: app/tenants/controller.go [hunk 3]
```diff
@@ -254,25 +246,16 @@ func ListOne(r *http.Request, cfg config.Config) (int, http.Header, []byte, erro
 		return code, h, output, err
```
Comment (pkoro) [inline app/tenants/controller.go]: Maybe error code here should be `404 Not found`?

File: app/tenants/controller.go [hunk 4]
```diff
@@ -347,30 +325,55 @@ func Update(r *http.Request, cfg config.Config) (int, http.Header, []byte, error
...
+		return code, h, output, err
 	}
 
+	// Check if name exists
+	sameName := []Tenant{}
+	filter = bson.M{"info.name": incoming.Info.Name}
+
+	err = mongo.Find(session, cfg.MongoDB.Db, "tenants", filter, "name", &sameName)
+
+	if len(sameName) > 1 {
```
Comment (pkoro) [inline app/tenants/controller.go]: Not sure if this check is needed here in the `Update` section. It should be ok, in principle, to use the same name during an update operation.

**Comment by pkoro** [comment]
+1

**Pull request merged**